integration/src/main.rs

```diff
@@ -205,11 +216,23 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
             // Create and execute program
             let mut linked_program = LinkedProgram::new(module, functions);
 
+            println!("args: {args:?}, ser: {serialized_args:?}");
             let mut stdin = StdIn::default();
             for arg in args {
                 let val = arg.parse::<u32>().unwrap();
                 stdin.write(&val);
             }
+
+            let serialized_inputs = serialized_args
```

Comment:
minor: `stdin.write_bytes` could be called for each element here (its panicking on error anyways), instead of having to read and keep all input files in memory at the same time before adding them to the input.

Reply:
do you mean pass the reader to write_bytes?